src/main/groovy/nebula/plugin/publishing/maven/MavenResolvedDependenciesPlugin.groovy

```diff
@@ -39,14 +40,19 @@ class MavenResolvedDependenciesPlugin implements Plugin<Project> {
                             def dependencies = xml.asNode()?.dependencies?.dependency
                             def dependencyMap = [:]
 
-                            dependencyMap['runtime'] = project.configurations.runtime.incoming.resolutionResult.allDependencies
+                            dependencyMap['runtime'] = project
+                                    .sourceSets.main.runtimeClasspath.sources
+                                    .findAll { it in Configuration }
+                                    .collect { it.incoming.resolutionResult.allDependencies }
+                                    .flatten()
+
                             dependencyMap['test'] = project.configurations.testRuntime.incoming.resolutionResult.allDependencies - dependencyMap['runtime']
                             dependencies?.each { Node dep ->
                                 def group = dep.groupId.text()
                                 def name = dep.artifactId.text()
                                 def scope = dep.scope.text()
 
-                                if (scope == 'provided') {
+                                if (scope == 'provided' || scope == 'compile') {
```

Comment:
Not sure if this is a good idea. New java-library will separate api (which need to be in compile scope) and implementation which should be in runtime scope.